### PR TITLE
Remote base add pronto protocol

### PR DESCRIPTION
**Pull request by cvwillegen**
# What does this implement/fix? 

Adds capability to send remote control commands specified with the Pronto protocol.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1563

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [X] ESP8266

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by cvwillegen** [comment]
I fixed a lot of linting errors. I'm not sure if I really need to fix those contants names, it makes the code differ a lot from the original...

Also, I don't know how to run ``black``. Could someone help me out here?

File: esphome/components/remote_base/pronto_protocol.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+/*
+ * Parse the string given as Pronto Hex, and send it a number of times given as argument.
+ */
+void ProntoProtocol::send_pronto_(RemoteTransmitData *dst, const std::vector<uint16_t> &data) {
+  if (data.size() < 4)
+    return;
+
+  unsigned int timebase = (MICROSECONDS_IN_SECONDS * data[1] + REFERENCE_FREQUENCY / 2) / REFERENCE_FREQUENCY;
```
Comment (glmnet) [inline esphome/components/remote_base/pronto_protocol.cpp]: Please use uintNN_t types

**Review by glmnet — COMMENTED** [review]
Looks good. just style review

File: esphome/components/remote_base/pronto_protocol.cpp (hunk 1 above)
Comment (cvwillegen) [reply esphome/components/remote_base/pronto_protocol.cpp]: Thanks for the pointer, updated!

**Review by jesserockz — APPROVED** [review]

**Pull request merged**